skills_ml/algorithms/occupation_classifiers/classifiers.py

```diff
@@ -1,141 +1,102 @@
-    """The Classifiers Object to classify each jobposting description to O*Net SOC code.
-
-    Example:
-
-    from airflow.hooks import S3Hook
-    from skills_ml.algorithms.occupation_classifiers.classifiers import Classifier
+import logging
+from collections import Counter, defaultdict
+import pickle
 
-    s3_conn = S3Hook().get_conn()
-    Soc = Classifier(s3_conn=s3_conn, classifier_id='ann_0614')
 
-    predicted_soc = Soc.classify(jobposting, mode='top')
+class SocClassifier(object):
+    """ Interface of SOC Code Classifier.
     """
-    def __init__(self, classifier_id='ann_0614', classifier=None,
-        s3_conn=None, s3_path='open-skills-private/model_cache/occupation_classifiers/', classify_kwargs=None, temporary_directory=None, **kwargs):
-        """Initialization of Classifier
-
-        Attributes:
-            classifier_id (str): classifier id
-            classifier_type (str): classifier type
-            s3_path (str): the path of the classifier on S3.
-            s3_conn (:obj: `boto.s3.connection.S3Connection`): the boto object to connect to S3.
-            files (:obj: `list` of (str)): classifier files need to be downloaded/loaded.
-            classifier (:obj): classifier object that will do the actually classification
-            classify_kwargs (:dict): arguments to pass through to the .classify method when called on a job posting
-        """
-        self.classifier_id = classifier_id
-        self.classifier_type = classifier_id.split('_')[0]
-        self.s3_conn = s3_conn
-        self.s3_path = s3_path + classifier_id
-        self.temporary_directory = temporary_directory or tempfile.TemporaryDirectory()
-        self.classifier = self._load_classifier(**kwargs) if classifier == None else classifier
-        self.classify_kwargs = classify_kwargs if classify_kwargs else {}
-
-    def _load_classifier(self, **kwargs):
-        if self.classifier_type == 'ann':
-            for f in list_files(self.s3_conn, self.s3_path):
-                filepath = os.path.join(self.temporary_directory, f)
-                if not os.path.exists(filepath):
-                    logging.warning('calling download from %s to %s', self.s3_path + f, filepath)
-                    download(self.s3_conn, filepath, os.path.join(self.s3_path, f))
-            ann_index = AnnoyIndexer()
-            ann_index.load(os.path.join(self.temporary_directory, self.classifier_id + '.index'))
-            return NearestNeighbors(s3_conn=self.s3_conn, indexer=ann_index, **kwargs)
-
-        elif self.classifier_type == 'knn':
-            return NearestNeighbors(s3_conn=self.s3_conn, indexed=False, **kwargs)
 
-        else:
-            print('Not implemented yet!')
-            return None
+    def __init__(self, classifier):
+        self.classifier = classifier
 
-    def classify(self, jobposting):
-        return self.classifier.predict_soc(jobposting, **(self.classify_kwargs))
+    def predict_soc(self, tokenized_list):
```

Comment:
Looking at all the code that was taken out and what's left, I'm not sure what the role of this class is. It doesn't transform anything or do any bootstrapping. It looks like anybody who can use this can just use the classifier_obj (`KNNDoc2VecClassifier`) directly with the same interface?

Reply:
yeah, my intention was to have an interface for `computer` to use. Maybe one would want to train their own classifier from scikit-learn. By instantiate a `SocClassifier(myClassifier)`, one can use that classifier with `computer`.